### DISCUSSION TITLE
Discord: Help to Set Up Different Series to Their Own Channels

### DISCUSSION THREAD

**Discussion by mornen** [jef/streetmerchant#1374] [Help]
I am trying to have each series alerts be in their own channel, and based on the dotenv file it appears this is supported.  I cannot get it to work properly, however, and all series are reported into all channels.  Has anyone done this succesfully?  What is put into DISCORD_NOTIFY_GROUP_3060TI= for example?  Here is a relevant copy of my dotenv:

DISCORD_NOTIFY_GROUP=
DISCORD_NOTIFY_GROUP_3060TI="<@xxxxxxxx9242>"
DISCORD_NOTIFY_GROUP_3070="<@xxxxxxxx1696>"
DISCORD_NOTIFY_GROUP_3080="<@xxxxxxxx5633>"
DISCORD_NOTIFY_GROUP_3090=
DISCORD_NOTIFY_GROUP_CORSAIR_SF=
DISCORD_NOTIFY_GROUP_RX6800=
DISCORD_NOTIFY_GROUP_RX6800XT=
DISCORD_NOTIFY_GROUP_RX6900XT=
DISCORD_NOTIFY_GROUP_RYZEN5600=
DISCORD_NOTIFY_GROUP_RYZEN5800=
DISCORD_NOTIFY_GROUP_RYZEN5900=
DISCORD_NOTIFY_GROUP_RYZEN5950=
DISCORD_NOTIFY_GROUP_SONYPS5C="<@xxxxxxx5150>"
DISCORD_NOTIFY_GROUP_SONYPS5DE="<@xxxxxxx5150>"
DISCORD_NOTIFY_GROUP_TEST=
DISCORD_WEB_HOOK="https://discordapp.com/api/webhooks/xxx9242/xxxxxxxxxx,https://discordapp.com/api/webhooks/xx1696/xxxxxxxxxxxxxx,https://discordapp.com/api/webhooks/xxxx5633/xxxxxxxxxxx,https://discordapp.com/api/webhooks/xxxxx5150/xxxxxxxxxxxx"

I have tried variations on this, such as channel names as groups.  I feel like I'm missing something obvious maybe but google hasn't helped.

Thanks

Edit:
I figured out part of the problem in a previously closed issue.  For the group id it needs to be formatted as <@&xxxxxxxxxxx>.  Also, it appears that this only mentions the user id provided.  It's possible I don't know enough about discord.  I am going to run everything into a single channel for now and if anyone has any thoughts I would still be interested in information or help.

My thoughts on the DISCORD_WEB_HOOK= allowing comma separated values was it would match the channel in the web hook to the id of the DISCORD_NOTIFIY_GROUP_XXXX= and post it in that channel.  If that is not how it works I think it would be a good feature to add.

Thanks again

**Comment by cyandura** [comment]
I want to know if this is possible. Can you have each product's notifications send to a different discord webhook? Ex. if a 3080 is in stock it sends a notification to a 3080 discord channel, and if a 3070 is in stock it sends to a separate 3070 discord channel, which would require a separate webhook. It doesn't seem like that's a supported feature? Then again I don't know a lot about discord either.

**Reply by tripp396** [reply]
I would love to see it possible to do different series in different channels. Right now I have to stack all series for RTX in one channel, all series for AMD in one channel, and all consoles in one channel. Running three instances really pegs my VM host.

**Comment by doghousedean** [comment]
I did it with docker, 3 different instances with different environment files

works a charm

**Reply by mornen** [reply]
Thanks for the replies.  After some more learning it seems maybe without docker, that the intention is to post in a single channel and use @mentions for each product.  Thereby limiting notifications to @mentions only, those members in the channel with the role matching DISCORD_NOTIFY_GROUP_3060TI= would get a notification.  This is still a theory, I haven't put it in use yet.  I did realize that I had no use for multiple webhooks, and have limited it to a single webhook now.
I'm trying to help my coworkers catch a new graphics card, and they're all on discord but trying keep people who want a 3080 from getting 3060ti notifications.  I'll have to read about docker more.  I have no experience with it.

**Reply by tripp396** [reply]
That’s how I eventually set it up. Gave everyone roles based on what they want and notify the roles.

**Reply by breacher18** [reply]
> That’s how I eventually set it up. Gave everyone roles based on what they want and notify the roles.

How did you implement this I tried to do the same thing but the mentions aren’t working. I’m trying to help a few friend get either a ps5 or Xbox so if I can get it to work it would be really helpful.

**Comment by breacher18** [comment]
I’ve tried this using roles and setting each individual model to mention the @role but it doesn’t work. The notification sends but the notification doesn’t seem to alert the role. Using the @everyone seems to be the only one that works correctly.

**Comment by PMVX** [comment]
I was trying to get this to work like a lot of discord channels having separate notification channels per product.. the discord webhook only allows to send to a single channel so to me it looks like if I'm tracking 10 products and want separate notifications for those products, that I would have to run the same number of instances .. so that means 10 separate threads with separate unique dotenv scanning for a single product and also need 10 discord hooks for each product. Is this correct?
